### PR TITLE
Containerize RSeQC-2.6.3 (resolves #114)

### LINKED ISSUE #114: Containerize: RSeQC

### PR DESCRIPTION
**Pull request by jvivian**
Uses HG38 Gencode.v23 BED file (baked into image).

### CODE REVIEW

File: rseqc-hg38-gencode23/test.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class TestRNASeQC(unittest.TestCase):
+
+    def test_docker_call(self):
+        self.assertEqual(0, docker_call(tool='quay.io/ucsc_cgl/rseqc-hg38-gencode23',
+                                        tool_parameters=[], work_dir=tempfile.gettempdir()))
+
+
+def docker_call(tool, tool_parameters, work_dir):
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/test.py:15]: Why the dupe?

File: rseqc-hg38-gencode23/rseqc.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+python $RSEQC_DIR/read_GC.py -i $INPUT -o $OUTDIR/$SAMPLE_NAME &
+
+# Read NVC
+python $RSEQC_DIR/read_NVC.py -i $INPUT -o $OUTDIR/$SAMPLE_NAME &
+
+# Read Quality
+python $RSEQC_DIR/read_quality.py -i $INPUT -o $OUTDIR/$SAMPLE_NAME &
+
+# RPKM Saturation
+python $RSEQC_DIR/RPKM_saturation.py -r $REF_GENE_BED -d '1++,1--,2+-,2-+' -i $INPUT -o $OUTDIR/$SAMPLE_NAME &
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/rseqc.sh:57]: Don't you need some sort of synchronization here? AFAIK, the shell kills all background jobs on exit.

File: rseqc-hg38-gencode23/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            R-base \
+            samtools \
+            python-pip \
+            python-numpy
+
+RUN pip install cython
+
+# Install and setup RSeQC
+COPY RSeQC-2.6.3.tar.gz /opt/
+RUN cd /opt && tar -zxf RSeQC-2.6.3.tar.gz
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/Dockerfile:17]: Remove tarball when done.

File: rseqc-hg38-gencode23/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+RUN cd /opt/RSeQC-2.6.3 && python setup.py install
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/Dockerfile:18]: Remove source directory when done.

File: rseqc-hg38-gencode23/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            python-dev \
+            R-base \
+            samtools \
+            python-pip \
+            python-numpy
+
+RUN pip install cython
+
+# Install and setup RSeQC
+COPY RSeQC-2.6.3.tar.gz /opt/
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/Dockerfile:16]: Why is this tarball part of the source tree? Is there no way to download it from an external source?

File: rseqc-hg38-gencode23/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,26 @@
+FROM ubuntu:14.04
+
+MAINTAINER John Vivian, jtvivian@gmail.com
+
+RUN apt-get update && apt-get install -y \
+            build-essential \
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/Dockerfile:6]: Too lazy to do a build container? build-essential pulls in a lot.

File: rseqc-hg38-gencode23/wrapper.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    chown -R ${user_id} /data
+}
+trap finish EXIT
+
+display_usage() {
+	echo "This script requires 2 arguments: path to BAM and UUID"
+	echo -e "\nUsage:\ndocker run -v <src:dst> quay.io/ucsc_cgl/rseqc-hg38-gencode23 /dst/BAM UUID \n"
+	}
+
+# if less than three arguments supplied, display usage
```
Comment (hannes-ucsc) [inline rseqc-hg38-gencode23/wrapper.sh:17]: Three? Also, this is pretty obvious …

File: rseqc-hg38-gencode23/rseqc.sh (hunk 2 above)
Comment (jvivian) [inline rseqc-hg38-gencode23/rseqc.sh:57]: I've tested this and all the python scripts get called in parallel and all produce output.

File: rseqc-hg38-gencode23/Dockerfile (hunk 4 above)
Comment (jvivian) [inline rseqc-hg38-gencode23/Dockerfile:18]: The python scripts actually get called from within that directory (see rseqc.sh, line 2). The `setup.py install` doesn't actually create any entrypoints or set anything on the PYTHONPATH, it compiles some of the included cython modules into importable .py files and checks for a couple of dependencies.

File: rseqc-hg38-gencode23/Dockerfile (hunk 5 above)
Comment (jvivian) [inline rseqc-hg38-gencode23/Dockerfile:16]: I spent about 30 minutes on this before giving up — The tarball is hosted on sourceforge which doesn't provide direct links only redirected mirrors.  `wget` is supposed to follow redirects, but in my tests I had `wget` hang about half the time it attempted to download the file.

File: rseqc-hg38-gencode23/Dockerfile (hunk 6 above)
Comment (jvivian) [inline rseqc-hg38-gencode23/Dockerfile:6]: good point, I'll address it.

**Comment by hannes-ucsc** [comment]
Underlying issue #114 was closed as `wont fix`

**Pull request closed without merging**